third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -812,46 +908,63 @@ func resourceSqlDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) e
 	d.Set("name", instance.Name)
 	d.Set("region", instance.Region)
 	d.Set("database_version", instance.DatabaseVersion)
-	d.Set("connection_name", instance.ConnectionName)
-	d.Set("service_account_email_address", instance.ServiceAccountEmailAddress)
+	<% unless version == 'ga' -%>
+	// Set external master config if there is any.
+	if instance.OnPremisesConfiguration != nil {
+		d.Set("source_ip_address", (strings.Split(instance.OnPremisesConfiguration.HostPort, ":"))[0])
+		d.Set("source_port", (strings.Split(instance.OnPremisesConfiguration.HostPort, ":"))[1])
+	} else {
```

Comment:
It's generally a good idea to set things even if they're empty, so I would probably have this `else` only contain `d.Set`s for the on prem properties and then have the other stuff get set unconditionally

Reply:
^

Reply:
I've tested - it doesn't work that way. Getting error:
```
Error: Error, failed to create instance tf-test-external-master-rmk1pcfeil: googleapi: Error 400: Missing parameter: Tier., required
```